dotnet/src/SemanticKernel.Abstractions/Security/TrustAwareString.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Security;
+
+/// <summary>
+/// A string wrapper that carries trust information.
+/// All field are readonly.
+/// </summary>
+public class TrustAwareString
+{
+    /// <summary>
+    /// Create a new empty trust aware string (default trusted).
+    /// </summary>
+    public static TrustAwareString Empty => new TrustAwareString(string.Empty, true);
+
+    /// <summary>
+    /// The raw string value.
+    /// </summary>
+    public string Value { get; }
+
+    /// <summary>
+    /// Whether the current value is trusted or not.
+    /// </summary>
+    public bool IsTrusted { get; }
+
+    /// <summary>
+    /// Create a new trust aware string.
+    /// </summary>
+    /// <param name="value">The raw string value</param>
+    /// <param name="isTrusted">Whether the raw string value is trusted or not</param>
+    public TrustAwareString(string value, bool isTrusted = true)
+    {
+        this.Value = value;
+        this.IsTrusted = isTrusted;
+    }
+
```

Comment:
I'm usually weary of this... not usually a fan of "magic tricks" in code... but this is a perfect place to use implicit operators. It'll make this much easier, transparent, to consume for those that don't care about trust state. Much like the [StringValues](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringvalues?view=dotnet-plat-ext-7.0) struct, which similarly can be treated just like a string unless you want to tap into its advanced (collection) features.

```suggestion
public static implicit operator string(TrustAwareString s) => s.ToString();
public static implicit operator TrustAwareString(string s) => new TrustAwareString(s);
```